sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -95,9 +95,7 @@ class MapboxMap internal constructor(
     onMapLoadErrorListener: OnMapLoadErrorListener? = null
   ) {
     initializeStyleLoad(onStyleLoaded, onMapLoadErrorListener)
-    handlerMain.post {
-      nativeMapWeakRef.call { (this as StyleManagerInterface).styleURI = styleUri }
-    }
+    nativeMapWeakRef.call { (this as StyleManagerInterface).styleURI = styleUri }
```

Comment:
A bit concerned here.
We're removing handler logic here but leaving it in `sdk/src/main/java/com/mapbox/maps/NativeObserver.kt`. If I remember correctly some some metrics benchmarks were having a deadlock because of not posting here.

Reply:
If tests pass - we can follow up later.

Reply:
agreed, let me dig up why we have that construct